src/modules/formBuilder/components/FormBuilderHeader.tsx

```diff
@@ -0,0 +1,80 @@
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import { Button, Grid, IconButton, Paper, Typography } from '@mui/material';
+import { Stack } from '@mui/system';
+import React, { useMemo } from 'react';
+import ButtonTooltipContainer from '@/components/elements/ButtonTooltipContainer';
+import { EditIcon } from '@/components/elements/SemanticIcons';
+import FormOptionsMenu from '@/modules/formBuilder/components/FormOptionsMenu';
+import { HmisEnums } from '@/types/gqlEnums';
+import { FormDefinitionFieldsForEditorFragment } from '@/types/gqlTypes';
+
+interface FormEditorHeaderProps {
+  formDefinition: FormDefinitionFieldsForEditorFragment;
+  lastUpdatedDate?: string | null;
+}
+
+const FormBuilderHeader: React.FC<FormEditorHeaderProps> = ({
+  formDefinition,
+  lastUpdatedDate,
+}) => {
+  const formRole = useMemo(
+    () => (formDefinition?.role ? HmisEnums.FormRole[formDefinition.role] : ''),
+    [formDefinition]
+  );
+
+  return (
+    <Paper sx={{ p: 4 }}>
+      <Stack direction='row' gap={1}>
+        <Typography sx={{ mb: 2 }} variant='h2'>
+          {formDefinition.title}
+        </Typography>
+        <ButtonTooltipContainer title='Edit Title'>
+          <IconButton
+            aria-label='edit title'
+            onClick={() => {}}
+            size='small'
+            sx={{ color: (theme) => theme.palette.links, mt: 0.25 }}
+          >
+            <EditIcon fontSize='inherit' />
+          </IconButton>
+        </ButtonTooltipContainer>
+      </Stack>
+      <Stack direction='row' sx={{ alignItems: 'center' }}>
+        <Grid container>
+          <Grid item xs={3}>
+            Form Identifier
+          </Grid>
```

Comment:
We could use the `CommonLabeledTextBlock` we usually use for this type of thing, so we're not introducing a new pattern of key/value display:

```
        <Grid container columnGap={4} rowGap={2}>
          <CommonLabeledTextBlock title='Form Identifier'>
            {formDefinition.identifier}
          </CommonLabeledTextBlock>
          <CommonLabeledTextBlock title='Form Type'>
            {formRole}
          </CommonLabeledTextBlock>
          <CommonLabeledTextBlock title='Status'>
            {formDefinition.status}
          </CommonLabeledTextBlock>
```